contracts/payloads/IGP84/PayloadIGP84.sol

```diff
@@ -0,0 +1,427 @@
+
+import {IFluidDex, IFluidAdminDex, IFluidDexResolver} from "../common/interfaces/IFluidDex.sol";
+
+import {IFluidVault, IFluidVaultT1} from "../common/interfaces/IFluidVault.sol";
+
+import {IFTokenAdmin, ILendingRewards} from "../common/interfaces/IFToken.sol";
+
+import {IDSAV2} from "../common/interfaces/IDSA.sol";
+import {IERC20} from "../common/interfaces/IERC20.sol";
+import {IProxy} from "../common/interfaces/IProxy.sol";
+import {PayloadIGPConstants} from "../common/constants.sol";
+import {PayloadIGPHelpers} from "../common/helpers.sol";
+import {PayloadIGPMain} from "../common/main.sol";
+
+contract PayloadIGP84 is PayloadIGPMain {
+    uint256 public constant PROPOSAL_ID = 84;
+
+    function execute() public virtual override {
+        super.execute();
+
+        // Action 1: Set USR-USDC and deUSD-USDC Max Supply Limit
+        action1();
+
+        // Action 2: Update oracles for USDC collateral vaults
+        action2();
+
+        // Action 3: Reset dust limits for ezETH-ETH DEX and ezETH<>wstETH T1 & ezETH-ETH<>wstETH T2 vaults
+        action3();
+
+    }
+
+    function verifyProposal() public view override {}
+
+    function _PROPOSAL_ID() internal view override returns (uint256) {
+        return PROPOSAL_ID;
+    }
+
+    /**
+     * |
+     * |     Proposal Payload Actions      |
+     * |__________________________________
+     */
+
+    // @notice Action 1: Set USR-USDC and deUSD-USDC Max Supply Limit
+    function action1() internal isActionSkippable(1) {
+        
+        {// USR-USDC DEX
+            address USR_USDC_DEX = getDexAddress(20);
+            {
+                // Decrease Max Supply Shares
+                IFluidDex(USR_USDC_DEX).updateMaxSupplyShares(
+                    7_500_000 * 1e18 // $15M
+            );
+            }
+        }
+
+        {// deUSD-USDC DEX
+            address deUSD_USDC_DEX = getDexAddress(19);
+            {
+                // Decrease Max Supply Shares
+                IFluidDex(deUSD_USDC_DEX).updateMaxSupplyShares(
+                    7_500_000 * 1e18 // $15M
+            );
+            }
+        }
+    }
+
+    // @notice Action 2: Update oracles for USDC collateral vaults
+    function action2() internal isActionSkippable(2) {
+        
+        {
+            address USDC_ETH_VAULT = getVaultAddress(100);
+    
+             IFluidVaultT1(USDC_ETH_VAULT).updateOracle(
+                    0x122bd7fbacef5d42cb178cba71df03d03d3e8821
+            );
+        }
+
+        {
+            address USDC_WBTC_VAULT = getVaultAddress(101);
+    
+             IFluidVaultT1(USDC_WBTC_VAULT).updateOracle(
+                    0x4446de5ba71dceeddec8715921199e9dadf55735
+            );
+        }
+
+        {
+            address USDC_cbBTC_VAULT = getVaultAddress(102);
+    
+             IFluidVaultT1(USDC_cbBTC_VAULT).updateOracle(
+                    0x4446de5ba71dceeddec8715921199e9dadf55735
+            );
```

Comment:
@bergben @xma8lu Same oracle for wbtc and cbbtc?

Reply:
yes, as CBBTC and WBTC is on debt side we both assume pegged to BTC, oracle simply is USDC / BTC